### ISSUE TITLE
[Bug]: [benchmark][cluster] Milvus search with an error"fail to Search, QueryNode ID=4, reason=QueryNode 9 can't serve, recovering: target node id not match target id"

### ISSUE THREAD

**Issue by jingkl** [milvus-io/milvus#21196]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.0-20221212-4d43df4f 
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.0dev4
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

deploy_config:
fouramf-server-cluster-8c16m-test
case_params:
fouramf-client-varchar-search-locust
other_params:
 --milvus_tag=2.2.0-20221212-4d43df4f -k test_concurrent_locust_custom_parameters -s --case_skip_clean_collection --deploy_mode=cluster --sync_report  --deploy_retain

sever:

```
fouramf-q4ldt-13-6484-etcd-0                                     1/1     Running     0              15m    10.104.4.169   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-etcd-1                                     1/1     Running     0              15m    10.104.5.10    4am-node12   <none>           <none>
fouramf-q4ldt-13-6484-etcd-2                                     1/1     Running     0              15m    10.104.9.252   4am-node14   <none>           <none>
fouramf-q4ldt-13-6484-milvus-datacoord-7787b655d6-hpzz8          1/1     Running     0              15m    10.104.4.159   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-milvus-datanode-8667fc9f6c-69cg8           1/1     Running     0              15m    10.104.4.163   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-milvus-indexcoord-67fd5f9f69-v7spn         1/1     Running     0              15m    10.104.4.161   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-milvus-indexnode-c6bc5c5cb-wwcm2           1/1     Running     0              15m    10.104.4.155   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-milvus-proxy-5db5f65f7f-nzqdv              1/1     Running     0              15m    10.104.4.156   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-milvus-querycoord-68844485d6-tl75s         1/1     Running     0              15m    10.104.4.158   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-milvus-querynode-84d888cbcd-nmzdh          1/1     Running     1 (10m ago)    15m    10.104.1.184   4am-node10   <none>           <none>
fouramf-q4ldt-13-6484-milvus-rootcoord-55bc55b474-blctw          1/1     Running     0              15m    10.104.1.183   4am-node10   <none>           <none>
fouramf-q4ldt-13-6484-minio-0                                    1/1     Running     0              15m    10.104.4.168   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-minio-1                                    1/1     Running     0              15m    10.104.5.12    4am-node12   <none>           <none>
fouramf-q4ldt-13-6484-minio-2                                    1/1     Running     0              15m    10.104.9.254   4am-node14   <none>           <none>
fouramf-q4ldt-13-6484-minio-3                                    1/1     Running     0              15m    10.104.1.194   4am-node10   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-bookie-0                            1/1     Running     0              15m    10.104.1.189   4am-node10   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-bookie-1                            1/1     Running     0              15m    10.104.4.175   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-bookie-2                            1/1     Running     0              15m    10.104.5.16    4am-node12   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-bookie-init-ffpzm                   0/1     Completed   0              15m    10.104.4.157   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-broker-0                            1/1     Running     0              15m    10.104.4.162   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-proxy-0                             1/1     Running     0              15m    10.104.5.251   4am-node12   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-pulsar-init-6mc97                   0/1     Completed   0              15m    10.104.4.160   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-recovery-0                          1/1     Running     0              15m    10.104.4.164   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-zookeeper-0                         1/1     Running     0              15m    10.104.4.170   4am-node11   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-zookeeper-1                         1/1     Running     0              14m    10.104.1.196   4am-node10   <none>           <none>
fouramf-q4ldt-13-6484-pulsar-zookeeper-2                         1/1     Running     0              13m    10.104.9.3     4am-node14   <none>           <none>
```

client log:

```
2022-12-13 09:04:15,312 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=1, message=fail to Search, QueryNode ID=4, reason=QueryNode 9 can't serve, recovering: target node id not match target id = 4, node id = 9)>, <Time:{'RPC start': '2022-12-13 09:04:14.008723', 'RPC error': '2022-12-13 09:04:15.312674'}> (decorators.py:108)
[2022-12-13 09:04:15,314 - ERROR - fouram]: Traceback (most recent call last):
  File "/src/fouram/client/util/api_request.py", line 33, in inner_wrapper
    res = func(*args, **kwargs)
  File "/src/fouram/client/util/api_request.py", line 70, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 610, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 469, in search
    return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 438, in _execute_search_requests
    raise pre_err
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 429, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to Search, QueryNode ID=4, reason=QueryNode 9 can't serve, recovering: target node id not match target id = 4, node id = 9)>
 (api_request.py:48)
```

### Steps To Reproduce

```markdown
1. create a collection 
        2. build index on vector column
        3. insert a certain number of vectors
        4. flush collection
        5. build index on vector column with the same parameters
        6. build index on on scalars column 
        7. count the total number of rows
        8. load collection
        9. perform concurrent operations
```

### Anything else?

```
'client': {'test_case_type': 'ConcurrentClientBase',
            'test_case_name': 'test_concurrent_locust_custom_parameters',
            'test_case_params': {'dataset_params': {'dim': 128,
                                                    'varchar_filled': True,
                                                    'scalars_index': ['float_1'],
                                                    'dataset_name': 'sift',
                                                    'dataset_size': '10w',
                                                    'ni_per': 50000,
                                                    'metric_type': 'L2'},
                                 'collection_params': {'other_fields': ['float_1']},
                                 'load_params': {},
                                 'search_params': {},
                                 'index_params': {'index_type': 'HNSW',
                                                  'index_param': {'M': 8,
                                                                  'efConstruction': 200}},
                                 'concurrent_params': {'concurrent_number': 20,
                                                       'during_time': '10m',
                                                       'interval': 20,
                                                       'spawn_rate': None},
                                 'concurrent_tasks': [{'type': 'search',
                                                       'weight': 10,
                                                       'params': {'nq': 10,
                                                                  'top_k': 10,
                                                                  'search_param': {'ef': 16},
                                                                  'expr': {'float_1': {'GT': -0.1,
                                                                                       'LT': 50000}},
                                                                  'random_data': True}}]}
```

**Comment by jiaoew1991** [comment]
It is panic by index out of range 
<img width="1111" alt="image" src="https://user-images.githubusercontent.com/2297455/207764409-429b0488-06fd-476f-9ae6-ba497fac817d.png">

**Comment by jiaoew1991** [comment]
It's a knowhere issue, which was fixed in Knowhere 1.3.6, please rerun it with pr https://github.com/milvus-io/milvus/pull/21241

**Comment by jingkl** [comment]
deploy_config：
fouramf-server-cluster-8c16m-test
case_params：
fouramf-client-varchar-search-locust
other_params：
--milvus_tag=2.2.0-20221216-1aa7a9a8 -k test_concurrent_locust_custom_parameters -s --case_skip_clean_collection --deploy_mode=cluster --sync_report --deploy_retain


image tag：2.2.0-20221216-1aa7a9a8
pymilvus： 2.2.1dev4
The problem did not reappear, turn off the issue
